datafusion/core/src/physical_planner.rs

```diff
@@ -605,6 +605,7 @@ impl DefaultPhysicalPlanner {
                         FileType::JSON => Arc::new(JsonFormat::default()),
                         FileType::AVRO => Arc::new(AvroFormat {} ),
                         FileType::ARROW => Arc::new(ArrowFormat {}),
+                        FileType::Extension(_ext) => return not_impl_err!("Extension FileTypes not supported in Copy To statements.")
```

Comment:
The built in Copy plans can't support external types (since we can't initialize them in DataFusion), but it should be not too difficult to build a custom COPY PhysicalPlan that uses a custom FileFormat, but otherwise is the same and relies on the build in Copy logical plan.

Reply:
Maybe we could have the `FileTypeExtension` have a method that returned `Arc<dyn FileFormat>` ?